apps/src/lib/metrics/MetricsReporter.ts

```diff
@@ -116,3 +115,5 @@ class MetricsReporter {
     );
   }
 }
+
+export default new MetricsReporter(new DashboardMetricsApi());
```

Comment:
For my understanding, is this a standard practice in typescript (define a class, then export a specific instance of it)? Is this sort of a singleton-ish pattern?

Reply:
Hmm, not sure if this a standard pattern in TS or not, but yeah, this is just so we can use this class as a singleton. And also, by doing so we're pre-selecting which version of the backend metrics api we want consumers to use (the dashboard one in this case).

Reply:
Yep! `MetricsReporter`'s constructor expects an argument of type `MetricsApi` (defined in `MetricsApi.ts`), which is stored as an instance member named `metricsApi`. And since `DashboardMetricsApi` implements `MetricsApi`, we can pass in an instance of `DashboardMetricsApi` here. And yep this was intentionally designed this way; this allows to create multiple different implementations of `MetricsApi` so the reporting code is agnostic of which backend we actually post metrics to. If we want to change which backend to post to, we just need to create the new class that implements `MetricsApi` and change this to `new MetricsReporter(new SomeOtherMetricsApi());`